telethon/network/__init__.py

```diff
@@ -7,5 +7,7 @@
 from .mtprotosender import MTProtoSender
 from .connection import (
     ConnectionTcpFull, ConnectionTcpIntermediate, ConnectionTcpAbridged,
-    ConnectionTcpObfuscated, ConnectionTcpMTProxy, ConnectionHttp
+    ConnectionTcpObfuscated, ConnectionTcpMTProxyAbridged,
+    ConnectionTcpMTProxyIntermediate,
+    ConnectionTcpMTProxyRandomizedIntermediate, ConnectionHttp, TcpMTProxy
```

Comment:
Is it correct that we cannot use MTProxy with, for example, `ConnectionTcpFull`? Otherwise another approach could be that any `Connection` supported a `MTProxy`.

Reply:
TcpFull is not supported in mtproto proxy:
https://github.com/seriyps/mtproto_proxy/blob/7b3e800dc93f52823d0e38d8817ed3aff88eac5f/src/mtp_obfuscated.erl#L147-L152
https://github.com/TelegramMessenger/MTProxy/blob/2c942119c4ee340c80922ba11d14fb3b10d5e654/net/net-tcp-rpc-ext-server.c#L226

Reply:
> TcpFull is not supported in mtproto proxy:
What about middleproxies?

Reply:
@cher-nov you mean the ones from this list https://core.telegram.org/getProxyConfig ? Yes, they use TcpFull, but I don't think telegram client should be able to connect to them (they use very different key exchange protocol and only intended to be used by mtproto proxies, not by clients AFAIK)